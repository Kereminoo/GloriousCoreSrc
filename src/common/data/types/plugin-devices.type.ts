import { PluginDeviceRecord } from "../records/plugin-device.record";

export type PluginDevices = 
{
    Mouse: PluginDeviceRecord[];
    Keyboard: PluginDeviceRecord[];
    valueE: PluginDeviceRecord[];
    MouseDock: PluginDeviceRecord[];
}