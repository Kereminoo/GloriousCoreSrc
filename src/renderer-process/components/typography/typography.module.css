.heading1 {
  font-family: var(--secondary-font-family);
  font-size: 24px;
  font-weight: 700;
  line-height: 32px;
  margin: 0;
}

.heading2 {
  font-family: var(--secondary-font-family);
  font-size: 20px;
  font-weight: 700;
  line-height: 24px;
  margin: 0;
}

.paragraphlargebold,
.paragraphlarge {
  font-family: var(--primary-font-family);
  font-size: 16px;
  line-height: 24px;
  margin: 0;
}

.paragraphmediumbold,
.paragraphmedium {
  font-family: var(--primary-font-family);
  font-size: 14px;
  line-height: 20px;
  margin: 0;
}

.paragraphsmallbold,
.paragraphsmall {
  font-family: var(--primary-font-family);
  font-size: 12px;
  line-height: 16px;
  margin: 0;
}

.paragraphxsmallbold,
.paragraphxsmall {
  font-family: var(--primary-font-family);
  font-size: 10px;
  line-height: 16px;
  margin: 0;
}

.paragraphlargebold,
.paragraphmediumbold,
.paragraphsmallbold,
.paragraphxsmallbold {
  font-weight: 600;
}

.buttonlarge {
  font-family: var(--primary-font-family);
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  margin: 0;
}

.buttonmedium {
  font-family: var(--secondary-font-family);
  font-size: 14px;
  font-weight: 600;
  line-height: 16px;
  margin: 0;
}

.buttonsmall {
  font-family: var(--secondary-font-family);
  font-size: 12px;
  font-weight: 600;
  line-height: 16px;
  margin: 0;
}

.linktextlarge {
  font-family: var(--secondary-font-family);
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
}

.linktextmedium {
  font-family: var(--secondary-font-family);
  font-size: 14px;
  font-weight: 600;
  line-height: 16px;
}

.linktextsmall {
  font-family: var(--secondary-font-family);
  font-size: 12px;
  font-weight: 500;
  line-height: 16px;
}

.primary {
  color: var(--base-shades-30);
}

.white {
  color: var(--base-shades-10);
}

.medium {
  color: var(--base-shades-30);
}

.bright {
  color: var(--base-shades-20);
}

.dark {
  color: var(--base-shades-90);
}

.underline {
  text-decoration: underline;
}
