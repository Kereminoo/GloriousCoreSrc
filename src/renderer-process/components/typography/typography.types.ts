import { Color } from '../component.types';

export type AllowedElements = 'h1' | 'h2' | 'h3' | 'p' | 'span';

export const textColorsOptions: (Color | undefined)[] = [
  Color.Base90,
  Color.Base30,
  Color.Base20,
  Color.Base10,
  undefined,
];

export enum TextType {
  Heading1 = 'heading1',
  Heading2 = 'heading2',
  ParagraphLargeBold = 'paragraphlargebold',
  ParagraphLarge = 'paragraphlarge',
  ParagraphMediumBold = 'paragraphmediumbold',
  ParagraphMedium = 'paragraphmedium',
  ParagraphSmallBold = 'paragraphsmallbold',
  ParagraphSmall = 'paragraphsmall',
  ParagraphXSmallBold = 'paragraphxsmallbold',
  ParagraphXSmall = 'paragraphxsmall',
  ButtonLarge = 'buttonlarge',
  ButtonMedium = 'buttonmedium',
  ButtonSmall = 'buttonsmall',
  LinkTextLarge = 'linktextlarge',
  LinkTextMedium = 'linktextmedium',
  LinkTextSmall = 'linktextsmall',
}

export type TextColor =
  | Color.Base100
  | Color.Base90
  | Color.Base60
  | Color.Base50
  | Color.Base40
  | Color.Base30
  | Color.Base20
  | Color.Base10
  | Color.Glorange60
  | Color.Glorange50
  | Color.Glorange20
  | Color.RedDark60
  | Color.GreenDark60;
