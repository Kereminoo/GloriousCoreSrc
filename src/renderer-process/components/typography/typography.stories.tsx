import { Meta, StoryObj } from '@storybook/react';
import Typography from './typography';
import { TextType, textColorsOptions } from './typography.types';
import React from 'react';
import { Color } from '../component.types';

const argTypes = {
  children: {
    description: 'The text content.',
  },
  type: {
    control: {
      type: 'select',
    },
    options: Object.values(TextType),
    description: 'The type of text to determine style and HTML element used.',
  },
  className: {
    description: 'The optional class to add to the element.',
  },
  element: {
    control: {
      type: 'select',
    },
    options: ['h1', 'h2', 'h3', 'p', 'span', undefined],
    description: 'The optional setting to change HTML element used.',
    defaultValue: {
      summary: 'different for each type',
    },
  },
  color: {
    control: {
      type: 'select',
    },
    options: textColorsOptions,
    description: 'The optional color settings for the text.',
    defaultValue: {
      summary: 'Color.Base30',
    },
  },
};

const meta: Meta<typeof Typography> = {
  title: 'Design System/Typography',
  component: Typography,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes,
};

export default meta;

type Story = StoryObj<typeof Typography>;

export const Default: Story = {
  args: {
    type: TextType.Heading1,
    children: 'This is a sample text',
    color: Color.Base30,
  },
};

export const AllText: Story = args => {
  return (
    <>
      {Object.values(TextType).map(type => (
        <React.Fragment key={type}>
          <Typography type={type as TextType} color={args.color}>
            {type}
          </Typography>
          <br />
        </React.Fragment>
      ))}
    </>
  );
};

AllText.args = {
  color: Color.Base30,
};
