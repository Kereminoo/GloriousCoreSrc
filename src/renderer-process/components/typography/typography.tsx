import React from 'react';
import styles from './typography.module.css';
import { AllowedElements, TextColor, TextType } from './typography.types';
import classNames from 'classnames';
import { Color } from '../component.types';

const typeToElementMapping: { [key in TextType]: AllowedElements } = {
  [TextType.Heading1]: 'h1',
  [TextType.Heading2]: 'h2',
  [TextType.ParagraphLargeBold]: 'p',
  [TextType.ParagraphLarge]: 'p',
  [TextType.ParagraphMediumBold]: 'p',
  [TextType.ParagraphMedium]: 'p',
  [TextType.ParagraphSmallBold]: 'p',
  [TextType.ParagraphSmall]: 'p',
  [TextType.ParagraphXSmallBold]: 'p',
  [TextType.ParagraphXSmall]: 'p',
  [TextType.ButtonLarge]: 'span',
  [TextType.ButtonMedium]: 'span',
  [TextType.ButtonSmall]: 'span',
  [TextType.LinkTextLarge]: 'span',
  [TextType.LinkTextMedium]: 'span',
  [TextType.LinkTextSmall]: 'span',
};

interface TypographyProps {
  children: React.ReactNode;
  type: TextType;
  className?: string;
  element?: AllowedElements;
  color?: TextColor;
  underline?: boolean;
  id?: string;
}

const Typography: React.FC<TypographyProps> = ({
  children,
  type,
  className,
  element = 'p',
  color = Color.Base30,
  underline = false,
  id,
}) => {
  const Element = element || typeToElementMapping[type];
  return (
    <Element
      {...(id && { id })}
      className={classNames(styles[type], className, { [styles.underline]: underline })}
      style={{ color: `var(${color})` }}
    >
      {children}
    </Element>
  );
};

export default Typography;
