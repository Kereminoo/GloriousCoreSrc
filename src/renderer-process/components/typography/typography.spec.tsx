// @ts-ignore: Unused variable
import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Typography from './typography';
import styles from './typography.module.css';
import { TextType } from './typography.types';
import { Color } from '../component.types';

describe('Typography', () => {
  it('renders correctly with type prop', () => {
    const { asFragment } = render(<Typography type={TextType.ParagraphMedium}>Test Text</Typography>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct element for given type', () => {
    render(<Typography type={TextType.ParagraphLarge}>Display Text</Typography>);
    const headingElement = screen.getByText('Display Text');
    expect(headingElement.tagName).toBe('P');
  });

  it('applies the correct class for a type', () => {
    render(<Typography type={TextType.Heading1}>Heading 1</Typography>);
    const headingElement = screen.getByText('Heading 1');
    expect(headingElement).toHaveClass(styles[TextType.Heading1]);
  });

  it('overrides the element type when element prop is provided', () => {
    render(
      <Typography type={TextType.Heading1} element="span">
        Heading as Span
      </Typography>
    );
    const spanElement = screen.getByText('Heading as Span');
    expect(spanElement.tagName).toBe('SPAN');
  });

  it('applies the correct color class', () => {
    render(
      <Typography type={TextType.Heading1} color={Color.Base20}>
        Danger Text
      </Typography>
    );
    const colorElement = screen.getByText('Danger Text');
    expect(colorElement).toHaveStyle(`color: var(${Color.Base20})`);
  });

  it('renders correctly without color prop', () => {
    const { asFragment } = render(<Typography type={TextType.Heading2}>No Color</Typography>);
    expect(asFragment()).toMatchSnapshot();
  });
});
