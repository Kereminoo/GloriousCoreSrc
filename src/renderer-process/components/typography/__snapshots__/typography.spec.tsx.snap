// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders correctly with type prop 1`] = `
<DocumentFragment>
  <p
    class="paragraphmedium"
  >
    Test Text
  </p>
</DocumentFragment>
`;

exports[`Typography renders correctly without color prop 1`] = `
<DocumentFragment>
  <p
    class="heading2"
  >
    No Color
  </p>
</DocumentFragment>
`;
